lib/connectors_async/job_store.rb

```diff
@@ -16,6 +17,8 @@ class JobNotFoundError < StandardError; end
     def initialize
       # multiple threads can write to the store
       @store = Concurrent::Hash.new
+      @watcher = ConnectorsAsync::JobWatcher.new(job_store: self)
+      @watcher.watch
```

Comment:
replace this call by `watch_immediatly: true` in the constructor

Reply:
It's surprisingly hard to test if it's done this way.

I also think it's better to keep constructor from having big side effects, such as spawning a new thread - makes it easier to maintain and read the code.